contracts/adapters/swap/drop/src/contract.rs

```diff
@@ -0,0 +1,320 @@
+    let checked_drop_factory_contract_address =
+        deps.api.addr_validate(&msg.drop_factory_contract_address)?;
+
+    let drop_factory_state: drop_factory::state::State = deps.querier.query_wasm_smart(
+        &checked_drop_factory_contract_address,
+        &drop_factory::msg::QueryMsg::State {},
+    )?;
+
+    // Store the drop core contract address
+    DROP_CORE_CONTRACT_ADDRESS.save(
+        deps.storage,
+        &deps.api.addr_validate(&drop_factory_state.core_contract)?,
+    )?;
+
+    let drop_core_config: drop_staking_base::state::core::Config = deps.querier.query_wasm_smart(
+        &checked_drop_factory_contract_address,
+        &drop_staking_base::msg::core::QueryMsg::Config {},
+    )?;
+
+    let canonical_denom = drop_core_config
+        .ld_denom
+        .ok_or(ContractError::CanonicalDenomNotSet {})?;
+
+    CANONICAL_DENOM.save(deps.storage, &canonical_denom)?;
+    BRIDGED_DENOM.save(deps.storage, &drop_core_config.remote_denom)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute(
+            "entry_point_contract_address",
+            checked_entry_point_contract_address.to_string(),
+        )
+        .add_attribute(
+            "drop_factory_contract_address",
+            checked_drop_factory_contract_address.to_string(),
+        )
+        .add_attribute(
+            "drop_core_contract_address",
+            drop_factory_state.core_contract,
+        )
+        .add_attribute("canonical_denom", canonical_denom)
+        .add_attribute("bridged_denom", drop_core_config.remote_denom))
+}
+
+///////////////
+/// EXECUTE ///
+///////////////
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> ContractResult<Response> {
+    match msg {
+        ExecuteMsg::Swap { operations } => execute_swap(deps, env, info, operations),
+        ExecuteMsg::TransferFundsBack {
+            swapper,
+            return_denom,
+        } => Ok(execute_transfer_funds_back(
+            deps,
+            env,
+            info,
+            swapper,
+            return_denom,
+        )?),
+        _ => {
+            unimplemented!()
+        }
+    }
+}
+
+fn execute_swap(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    _operations: Vec<SwapOperation>,
+) -> ContractResult<Response> {
+    // Get entry point contract address from storage
+    let entry_point_contract_address = ENTRY_POINT_CONTRACT_ADDRESS.load(deps.storage)?;
+
+    // Enforce the caller is the entry point contract
+    if info.sender != entry_point_contract_address {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Get coin in from the message info, error if there is not exactly one coin sent
+    let coin_in = one_coin(&info)?;
+
+    let bridged_denom = BRIDGED_DENOM.load(deps.storage)?;
+    let canonical_denom = CANONICAL_DENOM.load(deps.storage)?;
```

Comment:
This is vocabulary that's re-used from the lido satellite contract, is this name still accurate here or should it be `REMOTE_DENOM` and `BONDED_DENOM` or similar?

Reply:
Yes, REMOTE_DENOM and BONDED_DENOM are better, I will fix